File: articles/cascading-risk-and-systemic-decision-failure/python/run_all_cascading_risk_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Cascading Risk and Systemic Decision Failure."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "cascade_risk_score_model.py",
    "threshold_failure_model.py",
    "buffer_depletion_model.py",
    "common_mode_risk_model.py",
    "system_vulnerability_comparison.py",
    "decision_record_exporter.py",
    "cascading_risk_systemic_failure_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python cascading-risk workflows completed.")


if __name__ == "__main__":
    main()

File: articles/cascading-risk-and-systemic-decision-failure/r/run_all_cascading_risk_workflows.R
# run_all_cascading_risk_workflows.R
# Run all R workflows for Cascading Risk and Systemic Decision Failure.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "cascading_risk_systemic_failure_workflow.R",
  "system_profiles.R",
  "scenario_performance.R",
  "cascade_review_tables.R",
  "resilience_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R cascading-risk workflows completed.")
